Tasks/Common/azure-blobstorage-downloader/azureBlobStorageProvider.ts

```diff
@@ -0,0 +1,139 @@
+                    reject();
+                }
+            });
+            resolve(readStream);
+        });
+    }
+
+    private _ensureContainerExistence(): Promise<void> {
+        return new Promise<void>(async (resolve, reject) => {
+            if (!this._isContainerExists) {
+                var self = this;
+                this._blobSvc.createContainerIfNotExists(this._container, function (error, result, response) {
+                    if (!!error) {
+                        console.log("Failed to create container " + self._container + ". Error: " + error.message);
+                        reject(error);
+                    } else {
+                        self._isContainerExists = true;
+                        console.log("Created container " + self._container);
+                        resolve();
+                    }
+                });
+            } else {
+                resolve();
+            }
+        });
+    }
+
+    private _getItems(container: string, parentRelativePath?: string): Promise<models.ArtifactItem[]> {
+        var promise = new Promise<models.ArtifactItem[]>(async (resolve, reject) => {
+            var items: models.ArtifactItem[] = [];
+            await this._ensureContainerExistence();
+
+            var self = this;
+            this._blobSvc.listBlobsSegmented(container, null, (error, result, response) => {
+                if (!!error) {
+                    console.log("Failed to list items inside container: " + self._container + ". Error: " + error.message);
+                    reject(error);
+                } else {
+                    console.log("Successfully fetcted list of items.");
+                    items = self._convertBlobResultToArtifactItem(result.entries);
+                    resolve(items);
+                }
+            });
+        });
+
+        return promise;
+    }
+
+    private _convertBlobResultToArtifactItem(blobResult: azureStorage.BlobService.BlobResult[]): models.ArtifactItem[] {
+        var artifactItems: models.ArtifactItem[] = new Array<models.ArtifactItem>();
+        blobResult.forEach(element => {
+            var artifactitem: models.ArtifactItem = new models.ArtifactItem();
+            artifactitem.itemType = models.ItemType.File;
+            artifactitem.fileLength = parseInt(element.contentLength);
+            artifactitem.lastModified = new Date(element.lastModified + 'Z');
+            artifactitem.metadata = element.metadata;
+            artifactitem.path = element.name;
+            artifactItems.push(artifactitem);
+        });
+
+        return artifactItems;
+    }
+
+    private _storageAccount: string;
+    private _accessKey: string;
+    private _container: string;
+    private _prefixFolderPath: string;
+    private _isContainerExists: boolean = false;
+    private _blobSvc: azureStorage.BlobService;
+    private _maxListSize: number;
```

Comment:
how are you using maxListSize

Reply:
I initially thought of capping the number of blobs that can be downloaded, but later i removed that restriction. so now i am removing this variable, if needed we can cap tha number of blobs later on.